File: lab-report-api/src/main/java/com/ubo/labreport/model/Token.java
package com.ubo.labreport.model;

import com.ubo.labreport.enums.TokenType;
import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "t_token")
public class Token {

    @Id
    @GeneratedValue(generator = "UUID")
    private UUID id;

    @Column(unique = true)
    private String token;

    @Enumerated(EnumType.STRING)
    private TokenType tokenType = TokenType.BEARER;
    private Boolean expired;
    private Boolean revoked;

    @ManyToOne(fetch =FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}

File: lab-report-api/src/main/java/com/ubo/labreport/model/Token.kt
package com.ubo.labreport.model

import com.ubo.labreport.enums.TokenType
import jakarta.persistence.*
import org.hibernate.annotations.GenericGenerator

@Entity
data class Token(

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    val id : String,

    @Column(unique = true)
    val token: String,

    @Enumerated(EnumType.STRING)
    val tokenType: TokenType = TokenType.BEARER,
    var expired: Boolean,
    var revoked: Boolean,

    @ManyToOne(fetch =FetchType.LAZY)
    @JoinColumn(name = "user_id")
    val user: User
) {

  constructor(token: String, tokenType: TokenType, expired: Boolean, revoked: Boolean, user: User) : this(
      "",
      token = token,
      tokenType = tokenType,
      expired = expired,
      revoked = revoked,
      user = user
  )
}


